### PR TITLE
Use dynamic port generator for StartTests

### PR DESCRIPTION
**Pull request by liliankasem**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

Following up on PR feedback from my last [PR](https://github.com/Azure/azure-functions-core-tools/pull/3850), to address potential port issues:

https://github.com/Azure/azure-functions-core-tools/pull/3850#discussion_r1786874896

### CODE REVIEW

File: test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs [hunk 1]
```diff
@@ -76,6 +78,37 @@ public static void TryKillProcessForPort(string port)
             }
         }
 
+        public static int GetAvailablePort(int minPort = 7000, int maxPort = 8000)
+        {
+            for (int port = minPort; port <= maxPort; port++)
```
Comment (satvu) [inline test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs:83]: I'm assuming it won't take long to find an open port, but do we expect many ports to be taken up?

**Review by satvu — COMMENTED** [review]
Could you add a comment in the description to explain why this PR is needed? Guessing that we want to run multiple tests in parallel using different ports?

**Review by satvu — APPROVED** [review]

File: test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs (hunk 1 above)
Comment (liliankasem) [reply test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs:83]: There are only 20 tests so at most 20 ports will be taken up

File: test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs [hunk 2]
```diff
@@ -76,6 +78,37 @@ public static void TryKillProcessForPort(string port)
             }
         }
 
+        public static int GetAvailablePort(int minPort = 7000, int maxPort = 8000)
```
Comment (jviau) [inline test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs:81]: I saw a sample on stackoverflow that showed a way to get a random port by asking for a socket (or was it a TCP listener?) with a port of `0`, which lets the OS assign you a random port, then connecting to that socket to see what its port was.

Not sure if you are interested in that

Comment (liliankasem) [reply test/Azure.Functions.Cli.Tests/E2E/Helpers/ProcessHelper.cs:81]: Ohh that's cool, I will check it out

**Review by surgupta-msft — DISMISSED** [review]

**Pull request merged**